### PR TITLE
Add carried_totem.sc to survival

### PR DESCRIPTION
**Pull request by KingOfTheClouds**
Short script to mitigate the hand-icap of Totem of Undying (e.g. Hardcore world)

### CODE REVIEW

File: programs/survival/carried_totem.sc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Allow a Totem of Undying anywhere in the inventory to activate
+
+__config() -> ({ 'stay_loaded' -> true });
```
Comment (gnembon) [inline programs/survival/carried_totem.sc:3]: please leave config to specify for others if needed.

File: programs/survival/carried_totem.sc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+__on_player_takes_damage(player, amount, source, source_entity) -> (
+	totem = 'totem_of_undying';
+	if(player~'health' <= amount
+	&& source != 'outOfWorld'
+	&& player~'holds':0 != totem
+	&& query(player, 'holds', 'offhand'):0 != totem
+	&& inventory_remove(player, totem),
+		// borrow the offhand slot
+		if(offhand_stack = inventory_set(player, -1, 1, totem),
+			schedule(0, _(player, what, count, nbt) -> (
```
Comment (gnembon) [inline programs/survival/carried_totem.sc:14]: you could technically only use `_(outer(player), outer(offhand_stack)) -> inventory_set(player, offhand_stack......)`, but that works too.

File: programs/survival/carried_totem.sc (hunk 1 above)
Comment (KingOfTheClouds) [reply programs/survival/carried_totem.sc]: done

File: programs/survival/carried_totem.sc (hunk 2 above)
Comment (KingOfTheClouds) [reply programs/survival/carried_totem.sc]: rewritten with outer

**Review by gnembon — APPROVED** [review]

**Pull request merged**